src/graph/executor/query/FulltextIndexScanExecutor.cpp

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2022 vesoft inc. All rights reserved.
+//
+// This source code is licensed under Apache 2.0 License.
+
+#include "graph/executor/query/FulltextIndexScanExecutor.h"
+
+#include "common/datatypes/DataSet.h"
+#include "graph/planner/plan/Query.h"
+#include "graph/util/FTIndexUtils.h"
+
+using nebula::storage::StorageClient;
+using nebula::storage::StorageRpcResponse;
+using nebula::storage::cpp2::GetPropResponse;
+
+namespace nebula::graph {
+
+folly::Future<Status> FulltextIndexScanExecutor::execute() {
+  auto esAdapterResult = FTIndexUtils::getESAdapter(qctx_->getMetaClient());
+  if (!esAdapterResult.ok()) {
+    return esAdapterResult.status();
+  }
+  esAdapter_ = std::move(esAdapterResult).value();
+  auto* ftIndexScan = asNode<FulltextIndexScan>(node());
+  auto esQueryResult = accessFulltextIndex(ftIndexScan->index(), ftIndexScan->searchExpression());
+  if (!esQueryResult.ok()) {
+    LOG(ERROR) << esQueryResult.status().message();
+    return esQueryResult.status();
+  }
+  auto esResultValue = std::move(esQueryResult).value();
+  const auto& space = qctx()->rctx()->session()->space();
+  if (!isIntVidType(space)) {
```

Comment:
following getprops could reuse `getvertices/getedges` operators and fulltextindexscanexecutor only needs to fetch the vid/eid from ES.

Reply:
done